### DISCUSSION TITLE
Is it possible to change the data pin?

### DISCUSSION THREAD

**Discussion by darrenparkinson** [pixelit-project/PixelIt#249] [Q&A] [answered]
I have a 32x8 panel and I have a wemos plugged into a driver board which has a capacitor and a fuse etc. When used with the WLED software, I believe that uses pin d4 for the data connectivity which works fine with this board setup.  Is it possible to change the pin used for data via the web interface, or will I need to [change it in the code](https://github.com/pixelit-project/PixelIt/blob/5931417ab11bcda1abe27a871613621117d9a19a/src/PixelIt.ino#L85) -- is that ok to do?

**Comment by foorschtbar** [comment, marked as answer]
Yes, thats the right line of code. Change it to D4 and compile the code.

**Reply by darrenparkinson** [reply]
Great, thanks.  I updated that.  Had to jump through a few hoops to get there, but it's up and running 😀🎉

For future reference:
* I'm using VSCode so I used these instructions to install [Platform IO](https://docs.platformio.org/en/latest/integration/ide/vscode.html#installation). 
* I also hit [this issue](https://github.com/platformio/platformio-vscode-ide/issues/205)
* [This issue](https://github.com/platformio/platformio-vscode-ide/issues/3398) also helped me out
* Which I think led me [back here](https://docs.platformio.org/en/latest/faq/troubleshooting.html#faq-package-manager-error-5) which also helped.

Thanks again.

**Reply by darrenparkinson** [reply]
By way of follow up I found [this issue](https://github.com/pixelit-project/PixelIt/issues/79) quite useful since I was going to log a feature request to allow this to be selectable, but gather it's not easily possible.

**Reply by miccgn** [reply]
For performance reasons, the output pin is coded hard into the FastLED library which is being used for LED output. You basically would have to rewrite that library - and I assume the author of this library has good reasons for going that way (hard encoding rather than having a variable). The fact that the author of FastLED even provided a template parameter for this pin (which makes it very easy to change it during compile time) confirms that the author of the library thought about it.

An alternative idea would be to compile the library several times - for each pin that could be used - and include all those binaries within the software. However, that would not fit into the memory of an ESP8266.

To include: this feature request most likely won't happen.
